### PR TITLE
Add command `iobroker debug adapter` to start Node.js debug sessions

### PR DESCRIPTION
**Pull request by AlCalzone**
Fixes: #334 

```
iobroker debug <adapter> [--ip=<ip>] [--port=<port>] [--wait]
```
The base commands starts
```
node --inspect <adapter-main-file> --force --logs
```
--ip and --port change the default listen port for the debugger (e.g. to 0.0.0.0 for remote debugging)
--wait sets a breakpoint in the very first line.

### CODE REVIEW

File: lib/cli/cliCompact.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict';
+const CLI = require('./messages.js');
 const CLICommand = require('./cliCommand.js');
-const messages = require('./messages.js');
```
Comment (AlCalzone) [inline lib/cli/cliCompact.js:3]: @Apollon77 I just saw this and changed it to be in line with the other files here.

File: lib/cli/cliDebug.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                    path.join(adapterDir, 'io-package.json'),
+                    'utf8'
+                )
+            );
+        } catch (e) {
+            CLI.error.cannotLoadIoPackage(adapter);
+            return void callback(34);
+        }
+
+        const possibleMainFiles = [
```
Comment (foxriver76) [inline lib/cli/cliDebug.js:50]: package.json main attribute should also work

**Comment by Apollon77** [comment]
Maybe better use —debug as parametet which is the better choice since controller 2

File: lib/cli/cliDebug.js (hunk 2 above)
Comment (AlCalzone) [reply lib/cli/cliDebug.js]: added.

**Comment by AlCalzone** [comment]
Changed to --debug.
BTW: We should really make the resolution of the adapter main file a helper method in tools.js.

File: lib/cli/cliCompact.js (hunk 1 above)
Comment (Apollon77) [reply lib/cli/cliCompact.js:3]: Thank you!

**Comment by Apollon77** [comment]
Thank you!

**Pull request merged**